core/src/main/scala/sttp/tapir/Validator.scala

```diff
@@ -160,75 +200,130 @@ object Validator extends ValidatorMacros {
-  }
-  case class Any[T](validators: immutable.Seq[Validator[T]]) extends Validator[T] {
-    override def apply(t: T): List[ValidationError[_]] = {
-      val results = validators.map(_.apply(t))
-      if (results.exists(_.isEmpty)) {
-        List.empty
-      } else {
-        results.flatten.toList
+    override def show: Option[String] = validators.flatMap(_.show) match {
+      case Nil      => None
+      case x :: Nil => Some(x)
+      case xs       => Some(s"all(${xs.mkString(",")})")
+    }
+
+    override def apply(t: T): ValidationResult[T] = {
+      ValidationResult.partitionMap(
+        validators.toList.map(_.apply(t))
+      ) match {
+        case (Nil, _)     => ValidationResult.Valid(t)
+        case (invalid, _) => ValidationResult.Invalid(t, invalid.flatMap(_.errors))
       }
     }
 
-    override def contramap[TT](g: TT => T): Validator[TT] = if (validators.isEmpty) Any(Nil) else super.contramap(g)
-    override def or(other: Validator[T]): Validator[T] = if (validators.isEmpty) other else Any(validators :+ other)
+    override def contramap[TT](g: TT => T): Validator[TT] =
+      if (validators.isEmpty) All(Nil) else super.contramap(g)
+
+    override def and(other: Validator[T]): Validator[T] =
+      if (validators.isEmpty) other else All(validators :+ other)
   }
+  case class Any[T](validators: immutable.Seq[Validator[T]]) extends Validator[T] {
 
-  //
-
-  def show[T](v: Validator[T]): Option[String] = {
-    v match {
-      case Min(value, exclusive) => Some(s"${if (exclusive) ">" else ">="}$value")
-      case Max(value, exclusive) => Some(s"${if (exclusive) "<" else "<="}$value")
-      // TODO: convert to patterns when https://github.com/lampepfl/dotty/issues/12226 is fixed
-      case p: Pattern[T]                     => Some(s"~${p.value}")
-      case m: MinLength[T]                   => Some(s"length>=${m.value}")
-      case m: MaxLength[T]                   => Some(s"length<=${m.value}")
-      case m: MinSize[T, _]                  => Some(s"size>=${m.value}")
-      case m: MaxSize[T, _]                  => Some(s"size<=${m.value}")
-      case Custom(_, showMessage)            => showMessage.orElse(Some("custom"))
-      case Enumeration(possibleValues, _, _) => Some(s"in(${possibleValues.mkString(",")}")
-      case Mapped(wrapped, _)                => show(wrapped)
-      case All(validators) =>
-        validators.flatMap(show(_)) match {
-          case immutable.Seq()  => None
-          case immutable.Seq(s) => Some(s)
-          case ss               => Some(s"all(${ss.mkString(",")})")
-        }
-      case Any(validators) =>
-        validators.flatMap(show(_)) match {
-          case immutable.Seq()  => Some("reject")
-          case immutable.Seq(s) => Some(s)
-          case ss               => Some(s"any(${ss.mkString(",")})")
-        }
+    override def show: Option[String] = validators.flatMap(_.show) match {
```

Comment:
I think both designs are equivalent (abstract `show` method in the parent trait / `show` as a pattern-match), so I'm wondering what's the rationale here for the change?

Reply:
The idea was to keep the definition of validator in just one place. As you can see in this design when you define a Validator you can specify the error and the show(that contains a symbolic representation of the validator). 

I see the `show` as a property of the validator and not just an optional thing, this is why I centralized it. 
If this doesn't make sense for you I can use pattern matching no :)

Reply:
If validator wasn't a sealed trait I would definitely agree :) But since it is ... well I guess both designs are equivalent so I can go with either.